app/models/miq_server/worker_management/monitor.rb

```diff
@@ -146,6 +148,8 @@ def sync_needed?
       stop_apache                if roles_changed && !apache_needed?
       start_apache               if roles_changed &&  apache_needed?
 
+      EvmDatabase.restart_failover_monitor_service if roles_added.include?("database_operations") || roles_deleted.include?("database_operations")
```

Comment:
Perhaps `if (roles_added | roles_deleted).include?("database_operations")`

Sometimes I look for places to use array union/intersection to a fault.